app/ide-desktop/lib/dashboard/src/authentication/src/authentication/components/confirmRegistration.tsx

```diff
@@ -0,0 +1,62 @@
+/** @file Registration confirmation page for when a user clicks the confirmation link set to their
+ * email address. */
+import * as react from "react";
+import * as router from "react-router-dom";
+import toast from "react-hot-toast";
+
+import * as app from "../../components/app";
+import withRouter from "../../navigation";
+import * as auth from "../providers/auth";
+import * as loggerProvider from "../../providers/logger";
+
+// =================
+// === Constants ===
+// =================
+
+const REGISTRATION_QUERY_PARAMS = {
+  verificationCode: "verification_code",
+  email: "email",
+};
```

Comment:
consider adding `as const`.

also, not sure about how useful this is because it's only used in one place. but also i'm not sure if inlining these would violate style guide/convention

Reply:
also, consider merging with `RESET_PASSWORD_QUERY_PARAMS` and `LOGIN_QUERY_PARAMS`

Reply:
Style guide is strict about "no magic strings" so these need to be constants despite being used in only one place.

As for merging with those, it could be done but different pages use different params. It's useful to know which page uses which params. So I didn't merge them.